# I. Importation des compteurs Vélib / Fusion en un df / Observations

In [1]:
# Import library:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline 
import seaborn as sns

# Définition du chemin d'accès:
### Céline
%cd C:\Users\c.doussot\Desktop\Data Analyst\GitHub\ParisPyVelib_Datas
### Hermine

### Tarik
%cd C:\Users\Home\Documents\Git\ParisPyVelib\Data

[WinError 3] Le chemin d’accès spécifié est introuvable: 'C:\\Users\\c.doussot\\Desktop\\Data Analyst\\GitHub\\ParisPyVelib_Datas'
C:\Users\Home\Documents\Git\ParisPyVelib\ParisPyVelib-Celine
C:\Users\Home\Documents\Git\ParisPyVelib\Data


In [2]:
# Si Drive est utilisé:
    # from google.colab import drive
    # drive.mount('/content/drive') # montage de la connexion au serveur Drive Google
    # %cd /content/drive/MyDrive/ParisPyVelib/Notebook/Datas/ # définition du chemin d'accès:

In [3]:
# Import all df 
df_New = pd.read_csv('New_comptage-velo-donnees-compteurs.csv', sep = ';')
df_2020 = pd.read_csv('2020_comptage-velo-donnees-compteurs.csv', sep = ';')
df_2019 = pd.read_csv('2019_comptage-velo-donnees-compteurs.csv', sep = ';')
df_2018 = pd.read_csv('2018_comptage-velo-donnees-compteurs.csv', sep = ';')

In [4]:
# Création de df_2021 pour supprimer les doublons entre df_New et df_2020

### Ajout d'une colonne à df_New pour identifier les années de comptages présents
df_New['Année de comptage'] = df_New['Date et heure de comptage'].apply(lambda x : x[0:4])
print('df_New', df_New['Année de comptage'].unique()) # Pour vérifier qu'il y ait seulement 2020 et 2021

### Création de df_2021 en supprimant les données 2020
df_2021 = df_New[df_New['Année de comptage'] == '2021']
print('df_2021', df_2021['Année de comptage'].unique()) # Pour vérifier qu'il ne reste seulement 2021

### Suppression de la colonne 'Année de comptage' pour garder les même colonnes que les autres df:
df_2021 = df_2021.drop('Année de comptage', axis = 1)

df_New ['2020' '2021']
df_2021 ['2021']


In [5]:
# Create df that combine all df (col 'source' ajoutée pour identifier la provenance de la data)
df = pd.concat([df_2021.assign(Source='2021'), df_2020.assign(Source='2020'), 
                df_2019.assign(Source='2019'), df_2018.assign(Source='2018')])

# Afficher len des dfs et vérifier que len de df = somme des len des df
pd.DataFrame([{'df_2021':len(df_2021), 'df_2020':len(df_2020), 'df_2019':len(df_2019),'df_2018':len(df_2018),'df': len(df),
               'df = sum des dfs': len(df) == len(df_2021) + len(df_2020)+ len(df_2019)+ len(df_2018)}], index = ['len']).head()

df_2021  df_2020  df_2019  df_2018       df  df = sum des dfs
len   344986   784339   436729   157825  1723879              True

- La longeur total est >2 millions de lignes, le df créé a bien pris en compte tous les df

In [6]:
# Création d'une police pour soigner l'affichage dans print
c = '\033[1m'+'\033[4m'+'\033[95m' # Bolt + Underline + Purple
e = '\033[0m' # End

# Affichage des infos des dfs
print(c+'df_2021 :'+e), print(df_2021.info()) 
print('\n'+c+'df_2020 :'+e), print(df_2020.info())
print('\n'+c+'df_2019 :'+e), print(df_2019.info())
print('\n'+c+'df_2018 :'+e), print(df_2018.info())
print('\n'+c+'df :'+e), print(df.info()) # Si je mets df.info dans un même print que le label, 
                                            # le label apparaît après au lieu de avant je comprends pas pourquoi...

df_2021 :
<class 'pandas.core.frame.DataFrame'>
Int64Index: 344986 entries, 6561 to 947056
Data columns (total 9 columns):
 #   Column                                   Non-Null Count   Dtype  
---  ------                                   --------------   -----  
 0   Identifiant du compteur                  344986 non-null  object 
 1   Nom du compteur                          344986 non-null  object 
 2   Identifiant du site de comptage          344986 non-null  int64  
 3   Nom du site de comptage                  344986 non-null  object 
 4   Comptage horaire                         344986 non-null  float64
 5   Date et heure de comptage                344986 non-null  object 
 6   Date d'installation du site de comptage  344986 non-null  object 
 7   Lien vers photo du site de comptage      344986 non-null  object 
 8   Coordonnées géographiques                344986 non-null  object 
dtypes: float64(1), int64(1), object(7)
memory usage: 26.3+ MB
None

df_2020 :
<class 'pandas.co

(None, None)

**Observations:**
- Pas de NAN
- Pas le même nombre de données en fonction des années
    - pour 2021 c'est normal étant seulement sur 6 mois
    - Plus les années sont résentes, plus il y a de données (plus de compteurs ajoutés)
- pas de colonne Non-Null affiché pour df (je comprends pas pourquoi ???)
- Pas le même dtype en fonction des dfs pour 'Comptage horaire' (float64 ou int64)

In [7]:
# Nombre de modalités par colonne
print(c+'df_2021 :'+e, df_2021.nunique(), sep = '\n')
print(c,'df_2020 :'+e, df_2020.nunique(), sep = '\n')
print(c,'df_2019 :'+e, df_2019.nunique(), sep = '\n')
print(c,'df_2018 :'+e, df_2018.nunique(), sep = '\n')
print(c,'df :'+e, df.nunique(), sep = '\n')

df_2021 :
Identifiant du compteur                      97
Nom du compteur                              97
Identifiant du site de comptage              70
Nom du site de comptage                      70
Comptage horaire                            729
Date et heure de comptage                  3575
Date d'installation du site de comptage      36
Lien vers photo du site de comptage          70
Coordonnées géographiques                    70
dtype: int64

df_2020 :
Identifiant du compteur                      96
Nom du compteur                              96
Identifiant du site de comptage              69
Nom du site de comptage                      69
Comptage horaire                           1028
Date et heure de comptage                  8783
Date d'installation du site de comptage      35
Lien vers photo du site de comptage          69
Coordonnées géographiques                    69
dtype: int64

df_2019 :
Identifiant du compteur                      81
Nom du compteur               

**Observations:**
- Même nombre de modalités pour les colonnes dans chaque df d'origine mais différents pour le df créé:
    - 'Identifiant du compteur'/ 'Nom du compteur'
    - 'Identifiant du site de comptage'/ 'Nom du site de comptage'/ 'Lien vers photo du site de comptage'/'Coordonnées géographiques' 

In [8]:
# Affichage des coordonnées uniques de df pour identifier d'éventuelles erreurs
df2 = df.sort_values(by = ['Coordonnées géographiques'], axis=0)
for i in df2['Coordonnées géographiques'].unique():
    print(i)

# Affichage des noms de compteurs uniques de df pour identifier d'éventuelles erreurs
df2 = df.sort_values(by = ['Nom du compteur'], axis=0)
for i in df2['Nom du compteur'].unique():
    print(i)

48.82024,2.35902
48.82026,2.3592
48.82108,2.32537
48.82203,2.32545
48.82636,2.30303
48.82648,2.303149
48.826503,2.38434
48.82658,2.38409
48.82682,2.38465
48.829523,2.38699
48.8295233,2.38699
48.8303305218,2.40055083752
48.830331,2.400551
48.830449,2.353199
48.83068,2.35348
48.830901,2.353237
48.83421,2.26542
48.83436,2.377
48.8344028,2.376941
48.834695,2.332968
48.83511,2.33338
48.835143,2.333031
48.83521,2.33307
48.83848,2.37587
48.83977,2.30198
48.839883,2.267099
48.83992,2.26694
48.839927,2.267151
48.84015,2.26733
48.840801,2.333233
48.84201,2.36729
48.842091,2.301
48.84216,2.30115
48.84223,2.36811
48.843435,2.383378
48.846028,2.375429
48.846099,2.375456
48.84638,2.31529
48.848399,2.275932
48.8484,2.27586
48.85013,2.35423
48.851131,2.345678
48.851525,2.343298
48.85204,2.28589
48.85209,2.28508
48.85372,2.35702
48.85735,2.35211
48.860528,2.388364
48.86057,2.38886
48.860622,2.388724
48.86077,2.372742
48.86077,2.37305
48.860852,2.372279
48.86149,2.37376
48.86155,2.37407
48.86179,2.32014

**Observations:**
- Coordonnées: les valeurs ne sont pas arrondis au même niveau ce qui crée des doublons
    - Pour la suite il faudra faire les mêmes arrondis
- Noms de compteurs: beaucoup de doublons à cause d'écritures différentes (majuscule, espace...)

**Pour la suite il faut créer 4 col et en modifier 2 :**
- Modifier 'Nom du compteur' et 'Nom du site de comptage ' (pour tout mettre avec la même écriture et éviter les doublons)
- 'Direction' (indiquant le sens de circulation indiqué à la fin de 'Noms de compteurs')
- 'Longitude' (avec arrondi)
- 'Latitude' (avec arrondi)
- 'Coord' (reprenant longitude et latitude = coordonnées arrondies)

# Création de colonnes puis enregistrement du df obtenu

In [9]:
df.nunique()

Identifiant du compteur                      102
Nom du compteur                              147
Identifiant du site de comptage               71
Nom du site de comptage                      108
Comptage horaire                            1054
Date et heure de comptage                  29830
Date d'installation du site de comptage       37
Lien vers photo du site de comptage           76
Coordonnées géographiques                     90
Source                                         4
dtype: int64

In [10]:
# Modifier 'Nom du compteur' et 'Nom du site de comptage ' pour tout mettre à la même écriture

### Adresse en majuscule
df['Nom du compteur'] = df['Nom du compteur'].apply(lambda x : x.upper())
df['Nom du site de comptage'] = df['Nom du site de comptage'].apply(lambda x : x.upper())

### Remplacement de Boulevard par BD, Avenue par AV, HÔTEL par HOTEL
df['Nom du compteur'] = df['Nom du compteur'].apply(lambda x : ' '.join([item.replace('AVENUE','AV').replace('BOULEVARD','BD').replace('HÔTEL', 'HOTEL') for item in x.split()]))
df['Nom du site de comptage'] = df['Nom du site de comptage'].apply(lambda x : ' '.join([item.replace('AVENUE','AV').replace('BOULEVARD','BD').replace('HÔTEL', 'HOTEL') for item in x.split()]))

In [11]:
# Création de la col 'Direction' indiqué à la fin de 'Noms de compteurs'
df['Direction'] = df['Nom du compteur'].apply(lambda x : x.split(' ')[-1])
print(df['Direction'].value_counts()) # Vérification des modalités prises

# Création de la col 'Address' reprenant 'Nom de compteur' sans la direction à la fin
df['Nom du compteur'] = df['Nom du compteur'].replace('PONT DU GARIGLIANO SE-NO SE-NO', 'PONT DU GARIGLIANO SE-NO')
df['Address'] = df['Nom du compteur'].apply(lambda x : ' '.join([item for item in x.split()[0:-1]]))
list(df['Address'].unique())

NE-SO    273323
SO-NE    273201
SE-NO    265106
NO-SE    247445
E-O      186910
O-E      170132
S-N      160720
N-S      147042
Name: Direction, dtype: int64


['97 AV DENFERT ROCHEREAU',
 '105 RUE LA FAYETTE',
 '106 AV DENFERT ROCHEREAU',
 '100 RUE LA FAYETTE',
 '135 AV DAUMESNIL',
 '28 BD DIDEROT',
 '39 QUAI FRANÇOIS MAURIAC',
 "18 QUAI DE L'HOTEL DE VILLE",
 '27 BD DIDEROT',
 'VOIE GEORGES POMPIDOU',
 '67 BD VOLTAIRE',
 '10 AV DE LA GRANDE ARMÉE',
 '7 AV DE LA GRANDE ARMÉE',
 '72 BD VOLTAIRE',
 '89 BD DE MAGENTA',
 '16 AV DE LA PORTE DES TERNES',
 'PONT DU GARIGLIANO',
 '102 BD DE MAGENTA',
 '26 BD DE MÉNILMONTANT',
 'FACE AU 16 AV DE LA PORTE DES TERNES',
 '21 BD AUGUSTE BLANQUI',
 '72 BD RICHARD LENOIR',
 'PONT NATIONAL',
 'FACE AU 48 QUAI DE LA MARNE',
 '77 BD RICHARD LENOIR',
 "FACE 104 RUE D'AUBERVILLIERS",
 'FACE AU 70 QUAI DE BERCY',
 '6 RUE JULIA BARTET',
 "FACE AU 25 QUAI DE L'OISE",
 '35 BD DE MÉNILMONTANT',
 '152 BD DU MONTPARNASSE',
 '10 BD AUGUSTE BLANQUI',
 '243 BD SAINT GERMAIN',
 'TOTEM 64 RUE DE RIVOLI',
 '87 AV DE FLANDRE',
 '72 AV DE FLANDRE',
 '2 AV DE LA PORTE DE BAGNOLET',
 '44 AV DES CHAMPS ELYSÉES',
 "180 AV D'ITALI

In [12]:
# Création des colonnes 'longitude', et 'latitude'
df[['Latitude', 'Longitude']] = df['Coordonnées géographiques'].str.split(',', expand = True)

# Arrondis des coordonnées à 3 décimales
df['Latitude'] = df['Latitude'].astype('float64').round(3)
df['Longitude'] = df['Longitude'].astype('float64').round(3)

# Création de 'Coord'
df['Coord'] = df['Latitude'].astype('str') + ',' + df['Longitude'].astype('str')
df.nunique()

Identifiant du compteur                      102
Nom du compteur                              100
Identifiant du site de comptage               71
Nom du site de comptage                       76
Comptage horaire                            1054
Date et heure de comptage                  29830
Date d'installation du site de comptage       37
Lien vers photo du site de comptage           76
Coordonnées géographiques                     90
Source                                         4
Direction                                      8
Address                                       69
Latitude                                      37
Longitude                                     52
Coord                                         63
dtype: int64

**Note :**
- Pas facile de choisir le bon arrondis:
    - si arrondi >= 3, plusieurs coord correspondent à une même adresse
    - si arrondi < 3, plusieurs adresses associées aux mêmes coords
- Le meilleur compromis trouvé est d'arrondir à 3 décimales (on obtient alors 63 coordonnées pour x adresses, le nombre réel d'adresse étant <69 puisqu'il faut enlever les erreurs d'écritures restantes et les numéros de rue qui diffèrent avec la direction)
    - certains compteurs ont été remplacés avec le temps, dans ce cas les 2 compteurs = même Coord
    - certaines Coord englobent des compteurs N-S et S-N par exemple (pas génant pour la suite puisque identifié dans col Direction)

**Next Step:**
- Vérifier que chaque Coord correspond à une adresse unique (sans compter la direction)

In [13]:
# Identifier les coordonnées ayant les mêmes adresses:
df3 = df[['Coord', 'Address']].groupby(['Address','Coord']).count()
df3.head(60) # Pas possible tout observer d'un coup(73 lignes), il faut faire head(60) puis tail(13)
df3.tail(13)

Empty DataFrame
Columns: []
Index: [(PONT DE BERCY, 48.838,2.376), (PONT DE LA CONCORDE, 48.864,2.32), (PONT DES INVALIDES, 48.863,2.31), (PONT DES INVALIDES, 48.863,2.311), (PONT DU GARIGLIANO, 48.84,2.267), (PONT NATIONAL, 48.827,2.384), (PONT NATIONAL, 48.827,2.385), (QUAI D'ORSAY, 48.863,2.312), (TOTEM 64 RUE DE RIVOLI, 48.857,2.352), (TOTEM 73 BD DE SÉBASTOPOL, 48.864,2.351), (TOTEM 85 QUAI D'AUSTERLITZ, 48.842,2.367), (TOTEM COURS LA REINE, 48.865,2.314), (VOIE GEORGES POMPIDOU, 48.848,2.276)]

- 4 adresses possèdent 2 coordonnées différentes qu'il faut donc fusionner:
    - 3 AV DE LA PORTE D'ORLÉANS ('48.821,2.325':'48.822,2.325')
    - 36 QUAI DE GRENELLE ('48.852,2.285':'48.852,2.286')
    - PONT DES INVALIDES ('48.863,2.31':'48.863,2.311')
    - PONT NATIONAL ('48.827,2.384': '48.827,2.385')

In [14]:
# Fusion des 4 Coords (étant la même adresse)
df['Coord'] = df['Coord'].replace({'48.821,2.325':'48.822,2.325', '48.852,2.285':'48.852,2.286', 
                                   '48.863,2.31':'48.863,2.311','48.827,2.384': '48.827,2.385'})

# Identifier les adresses ayant les même coordonnées
df2 = df[['Coord', 'Address']].groupby(['Coord', 'Address']).count()
df2.head(59) # Pas possible tout observer d'un coup(69 lignes), il faut faire head(59) puis tail(10)
df2.tail(10)

Empty DataFrame
Columns: []
Index: [(48.885,2.327, 20 AV DE CLICHY), (48.889,2.375, 72 AV DE FLANDRE), (48.889,2.375, 72 AV FLANDRE), (48.889,2.375, 87 AV DE FLANDRE), (48.89,2.369, FACE 104 RUE D'AUBERVILLIERS), (48.891,2.385, FACE AU 25 QUAI DE L'OISE), (48.891,2.386, FACE AU 48 QUAI DE LA MARNE), (48.896,2.36, 72 RUE DE LA CHAPELLE), (48.897,2.345, 69 BD ORNANO), (48.897,2.346, 74 BD ORNANO)]

- Chaque coord correspond bien à une même localisation, seul le sens de circulation peut changer (ce qui explique un numéro de rue parfois différent)
- 9 coordonnées correspondent à la même adresse (numéro de rue différent avec la direction):
    - BD AUGUSTE BLANQUI '48.83,2.353':'48.831,2.353'
    - BD RICHARD LENOIR '48.861,2.372':'48.861,2.373'
    - BD VOLTAIRE '48.861,2.374':'48.862,2.374'
    - BD DE MENILMONTANT '48.861,2.388':'48.861,2.389'
    - AV DE LA PORTE DE BAGNOLET '48.865,2.409':'48.865,2.41'
    - RUE LA FAYETTE '48.877,2.35':'48.878,2.351'
    - BD DE MAGENTA '48.878,2.354':'48.878,2.355'
    - AV DE LA PORTE DES TERNES '48.882,2.281':'48.882,2.282'
    - BD ORNANO '48.897,2.345':'48.897,2.346'

In [15]:
# Fusion des 9 Coords (étant la même adresse)
df['Coord'] = df['Coord'].replace({'48.83,2.353':'48.831,2.353','48.861,2.372':'48.861,2.373','48.861,2.374': '48.862,2.374',
                                   '48.861,2.388': '48.861,2.389','48.865,2.409': '48.865,2.41', '48.877,2.35':'48.878,2.351',
                                  '48.878,2.354':'48.878,2.355','48.882,2.281':'48.882,2.282','48.897,2.345':'48.897,2.346'})

# Vérification finale que chaque coordonée = une même adresse peut importe le sens
df2 = df[['Coord', 'Direction', 'Address']].groupby(['Coord', 'Direction', 'Address']).count()
df2.head(59) # Pas possible tout observer d'un coup(100 lignes), il faut faire head(59) puis tail(41)
df2.tail(41)

Empty DataFrame
Columns: []
Index: [(48.863,2.311, N-S, PONT DES INVALIDES), (48.863,2.311, S-N, PONT DES INVALIDES), (48.863,2.312, E-O, QUAI D'ORSAY), (48.863,2.312, O-E, QUAI D'ORSAY), (48.864,2.32, N-S, PONT DE LA CONCORDE), (48.864,2.32, S-N, PONT DE LA CONCORDE), (48.864,2.351, N-S, TOTEM 73 BD DE SÉBASTOPOL), (48.864,2.351, S-N, TOTEM 73 BD DE SÉBASTOPOL), (48.865,2.314, E-O, TOTEM COURS LA REINE), (48.865,2.314, O-E, TOTEM COURS LA REINE), (48.865,2.354, NE-SO, 38 RUE TURBIGO), (48.865,2.354, SO-NE, 38 RUE TURBIGO), (48.865,2.41, E-O, FACE AU 4 AV DE LA PORTE DE BAGNOLET), (48.865,2.41, O-E, 2 AV DE LA PORTE DE BAGNOLET), (48.865,2.41, O-E, FACE AU 4 AV DE LA PORTE DE BAGNOLET), (48.87,2.307, NO-SE, 33 AV DES CHAMPS ELYSÉES), (48.87,2.307, SE-NO, 44 AV DES CHAMPS ELYSÉES), (48.875,2.292, NO-SE, 7 AV DE LA GRANDE ARMÉE), (48.875,2.292, SE-NO, 10 AV DE LA GRANDE ARMÉE), (48.878,2.351, E-O, 105 RUE LA FAYETTE), (48.878,2.351, O-E, 100 RUE LA FAYETTE), (48.878,2.355, NO-SE, 89 BD DE MAGENTA), (48.878,2.355, SE-NO, 102 BD DE MAGENTA), (48.878,2.355, SE-NO, 102 BD MAGENTA), (48.882,2.282, E-O, 16 AV DE LA PORTE DES TERNES), (48.882,2.282, O-E, FACE AU 16 AV DE LA PORTE DES TERNES), (48.885,2.327, NO-SE, 20 AV DE CLICHY), (48.885,2.327, SE-NO, 20 AV DE CLICHY), (48.889,2.375, NE-SO, 87 AV DE FLANDRE), (48.889,2.375, SO-NE, 72 AV DE FLANDRE), (48.889,2.375, SO-NE, 72 AV FLANDRE), (48.89,2.369, N-S, FACE 104 RUE D'AUBERVILLIERS), (48.89,2.369, S-N, FACE 104 RUE D'AUBERVILLIERS), (48.891,2.385, NE-SO, FACE AU 25 QUAI DE L'OISE), (48.891,2.385, SO-NE, FACE AU 25 QUAI DE L'OISE), (48.891,2.386, NE-SO, FACE AU 48 QUAI DE LA MARNE), (48.891,2.386, SO-NE, FACE AU 48 QUAI DE LA MARNE), (48.896,2.36, N-S, 72 RUE DE LA CHAPELLE), (48.896,2.36, S-N, 72 RUE DE LA CHAPELLE), (48.897,2.346, N-S, 69 BD ORNANO), (48.897,2.346, S-N, 74 BD ORNANO)]

- Chaque coordonnées correspond bien à une adresse ayant pour la plupart 2 sens de circulation
- 5 coordonnées possèdent une seule direction de compteur:
    - 3 AV DE LA PORTE D'ORLÉANS '48.822,2.325'
    - 135 AV DAUMESNIL '48.843,2.383'
    - 30 RUE SAINT JACQUES '48.851,2.346'
    - 21 BD SAINT MICHEL '48.852,2.343'
    - 243 BD SAINT GERMAIN '48.862,2.32'

In [16]:
# Mise à jours des col 'latitude' et 'longitude'
df[['Latitude', 'Longitude']] = df['Coord'].str.split(',', expand = True)

# Affichage des coordonnées à 5 décimales (nécessaire pour la suite lors d'un affichage map)
df['Latitude'] = df['Latitude'].astype('float64').apply(lambda x: format(x,".5f"))
df['Longitude'] = df['Longitude'].astype('float64').apply(lambda x: format(x,".5f"))

# Mise à jours de la col 'Coord'
df['Coord'] = df['Latitude'].astype('str') + ',' + df['Longitude'].astype('str')

# Création d'une colonne 'Adress_full' qui reprend les coord avec les directions
df['Coord_full'] = df['Coord'] + '-' + df['Direction']

df.nunique() # On a maintenant 50 coordonnées correspondant toutes bien à des adresses différentes

Identifiant du compteur                      102
Nom du compteur                              100
Identifiant du site de comptage               71
Nom du site de comptage                       76
Comptage horaire                            1054
Date et heure de comptage                  29830
Date d'installation du site de comptage       37
Lien vers photo du site de comptage           76
Coordonnées géographiques                     90
Source                                         4
Direction                                      8
Address                                       69
Latitude                                      35
Longitude                                     43
Coord                                         50
Coord_full                                    95
dtype: int64

In [17]:
# Mettre en forme date-time et ajout de colonnes pour séparer date de comptage

### Mettre en forme date-time les col 'Date et heure de comptage' et 'Date d'installation du site de comptage'
df['Date et heure de comptage']= df['Date et heure de comptage'].str.split('+', expand= True)
df['Date et heure de comptage'] = pd.to_datetime(df['Date et heure de comptage'], format = '%Y-%m-%dT%H:%M:%S')
df['''Date d'installation du site de comptage'''] = pd.to_datetime(df['''Date d'installation du site de comptage'''], format = '%Y-%m-%d')

# Créer des col pour séparer date de comptage 
df['Y_Date_Count'] = df['Date et heure de comptage'].dt.year
df['M_Date_Count'] = df['Date et heure de comptage'].dt.month
df['D_Date_Count'] = df['Date et heure de comptage'].dt.day
df['Dweek_Date_Count'] = df['Date et heure de comptage'].dt.weekday # va de 0 lundi à 6 dimanche
df['H_Date_Count'] = df['Date et heure de comptage'].dt.hour

# Créer des col pour séparer date d'installation 
df['Y_Date_Instal'] = df['''Date d'installation du site de comptage'''].dt.year
df['M_Date_Instal'] = df['''Date d'installation du site de comptage'''].dt.month
df['D_Date_Instal'] = df['''Date d'installation du site de comptage'''].dt.day

In [18]:
# Rennomer les col pour simplifier et supprimer les espaces
df = df.rename({'Identifiant du compteur': 'Id_old',
                'Nom du compteur': 'Address1_old',
                'Lien vers photo du site de comptage': 'Photo_old',
                'Coordonnées géographiques': 'Coord_old',
                'Identifiant du site de comptage': 'Id',
                'Nom du site de comptage':'Address2_old',
                'Comptage horaire':'Count_by_hour',
                'Date et heure de comptage':'Date_count', 
                '''Date d'installation du site de comptage''':'Date_instal'}, axis = 1)

In [19]:
df = df.sort_values(by = 'Date_count')
df.index = range(len(df.index))
# df = df.reset_index(drop= False)
df.head()

Id_old                       Address1_old         Id  \
0  100042374-110042374        VOIE GEORGES POMPIDOU NE-SO  100042374   
1  100042374-109042374        VOIE GEORGES POMPIDOU SO-NE  100042374   
2         100003098-SC     106 AV DENFERT ROCHEREAU NE-SO  100003098   
3         100003097-SC             105 RUE LA FAYETTE E-O  100003097   
4  100036719-103036719  18 QUAI DE L'HOTEL DE VILLE SE-NO  100036719   

                     Address2_old  Count_by_hour Date_count Date_instal  \
0           VOIE GEORGES POMPIDOU            1.0 2018-01-01  2017-12-15   
1           VOIE GEORGES POMPIDOU            0.0 2018-01-01  2017-12-15   
2  106 AV DENFERT ROCHEREAU NE-SO            3.0 2018-01-01  2012-02-22   
3          105 RUE LA FAYETTE E-O            8.0 2018-01-01  2012-02-13   
4     18 QUAI DE L'HOTEL DE VILLE            3.0 2018-01-01  2017-07-12   

                                           Photo_old           Coord_old  \
0  https://www.eco-visio.net/Photos/100042374/151...  48.848399,2.275932   
1  https://www.eco-visio.net/Photos/100042374/151...  48.848399,2.275932   
2  https://www.eco-visio.net/Photos/100003098/133...  48.835143,2.333031   
3  https://www.eco-visio.net/Photos/100003097/157...  48.877667,2.350556   
4  https://www.eco-visio.net/Photos/100036719/149...    48.85372,2.35702   

  Source  ...             Coord              Coord_full Y_Date_Count  \
0   2018  ...  48.84800,2.27600  48.84800,2.27600-NE-SO         2018   
1   2018  ...  48.84800,2.27600  48.84800,2.27600-SO-NE         2018   
2   2018  ...  48.83500,2.33300  48.83500,2.33300-NE-SO         2018   
3   2018  ...  48.87800,2.35100    48.87800,2.35100-E-O         2018   
4   2018  ...  48.85400,2.35700  48.85400,2.35700-SE-NO         2018   

  M_Date_Count D_Date_Count Dweek_Date_Count  H_Date_Count  Y_Date_Instal  \
0            1            1                0             0           2017   
1            1            1                0             0           2017   
2            1            1                0             0           2012   
3            1            1                0             0           2012   
4            1            1                0             0           2017   

   M_Date_Instal  D_Date_Instal  
0             12             15  
1             12             15  
2              2             22  
3              2             13  
4              7             12  

[5 rows x 24 columns]

In [20]:
#Affichage du nombre de modalité par colonne du nouveau df
df.nunique()

Id_old                102
Address1_old          100
Id                     71
Address2_old           76
Count_by_hour        1054
Date_count          29830
Date_instal            37
Photo_old              76
Coord_old              90
Source                  4
Direction               8
Address                69
Latitude               35
Longitude              43
Coord                  50
Coord_full             95
Y_Date_Count            4
M_Date_Count           12
D_Date_Count           31
Dweek_Date_Count        7
H_Date_Count           24
Y_Date_Instal           6
M_Date_Instal           9
D_Date_Instal          21
dtype: int64

In [21]:
df.head()

Id_old                       Address1_old         Id  \
0  100042374-110042374        VOIE GEORGES POMPIDOU NE-SO  100042374   
1  100042374-109042374        VOIE GEORGES POMPIDOU SO-NE  100042374   
2         100003098-SC     106 AV DENFERT ROCHEREAU NE-SO  100003098   
3         100003097-SC             105 RUE LA FAYETTE E-O  100003097   
4  100036719-103036719  18 QUAI DE L'HOTEL DE VILLE SE-NO  100036719   

                     Address2_old  Count_by_hour Date_count Date_instal  \
0           VOIE GEORGES POMPIDOU            1.0 2018-01-01  2017-12-15   
1           VOIE GEORGES POMPIDOU            0.0 2018-01-01  2017-12-15   
2  106 AV DENFERT ROCHEREAU NE-SO            3.0 2018-01-01  2012-02-22   
3          105 RUE LA FAYETTE E-O            8.0 2018-01-01  2012-02-13   
4     18 QUAI DE L'HOTEL DE VILLE            3.0 2018-01-01  2017-07-12   

                                           Photo_old           Coord_old  \
0  https://www.eco-visio.net/Photos/100042374/151...  48.848399,2.275932   
1  https://www.eco-visio.net/Photos/100042374/151...  48.848399,2.275932   
2  https://www.eco-visio.net/Photos/100003098/133...  48.835143,2.333031   
3  https://www.eco-visio.net/Photos/100003097/157...  48.877667,2.350556   
4  https://www.eco-visio.net/Photos/100036719/149...    48.85372,2.35702   

  Source  ...             Coord              Coord_full Y_Date_Count  \
0   2018  ...  48.84800,2.27600  48.84800,2.27600-NE-SO         2018   
1   2018  ...  48.84800,2.27600  48.84800,2.27600-SO-NE         2018   
2   2018  ...  48.83500,2.33300  48.83500,2.33300-NE-SO         2018   
3   2018  ...  48.87800,2.35100    48.87800,2.35100-E-O         2018   
4   2018  ...  48.85400,2.35700  48.85400,2.35700-SE-NO         2018   

  M_Date_Count D_Date_Count Dweek_Date_Count  H_Date_Count  Y_Date_Instal  \
0            1            1                0             0           2017   
1            1            1                0             0           2017   
2            1            1                0             0           2012   
3            1            1                0             0           2012   
4            1            1                0             0           2017   

   M_Date_Instal  D_Date_Instal  
0             12             15  
1             12             15  
2              2             22  
3              2             13  
4              7             12  

[5 rows x 24 columns]

In [22]:
# Enregister le nouveau df sous format csv
df.to_csv('2018-2021_comptage-velo-donnees-compteurs.csv')